### PR TITLE
Shared String Dictionary ctd.

### PR DESCRIPTION
**Pull request by Tyler-IN**
the great refactoring

also fixes integration tests

also fixes integration tests logging on windows

also makes integration tests just log to the integration test logs instead of a file

but it's a hack - can be refactored out of scope of this PR

### CODE REVIEW

**Comment by Tyler-IN** [comment]
Preferaby, Func\<ILogHandler\> can be replaced w/ IoC in a later PR.

**Comment by Tyler-IN** [comment]
should be no cases of 'new byte[]' in the robust serializer & associated classes, and string alloc in unpacking is limited to max string length for dict (420) so no more 64k buffer alloc'd there

File: Robust.Server/Maps/MapLoader.cs [hunk 1]
```diff
@@ -99,6 +101,8 @@ public void SaveBlueprint(GridId gridId, string yamlPath)
 
                 var data = new MapData(reader);
 
+                LoadedMapData?.Invoke(data.Stream, resPath.ToRelativeSystemPath());
```
Comment (PJB3005) [inline Robust.Server/Maps/MapLoader.cs]: System path should not be exposed like this.

**Review by PJB3005 — COMMENTED** [review]
So this is interesting. Apparently `server_config.toml` already has a definition for `log.enabled = false` and this is causing it to be disabled. This isn't an issue on its own but I did notice that when logging is disabled like this the log file still gets *created*, but it stays empty.

File: Robust.Server/Maps/MapLoader.cs (hunk 1 above)
Comment (Tyler-IN) [reply Robust.Server/Maps/MapLoader.cs]: Honestly I have no idea why it's got System in the name.
"/\Maps\saltern.yml" is relative to the resource root.

**Comment by Tyler-IN** [comment]
So should I remove `log.enabled` handling?

**Comment by PJB3005** [comment]
Since you provided a suitable alternative for the integration tests I guess that could work, yeah.

**Pull request merged**